### PR TITLE
feat: [DetailsV2] trait bubbles

### PR DESCRIPTION
**Pull request by JackShort**
** adds trait chips for the details page
** note: most of the data is not available - like top trait we are currently not getting and we don't have images for the traits

figma: https://www.figma.com/file/jG2aAlirIlv3uEtMIGUpo1/%F0%9F%8F%99%EF%B8%8F-Detail-Page-v2?type=design&node-id=1686-103603&t=htzsIAkjm0qHk5RD-0

### Automated testing

- [x] Unit test

### CODE REVIEW

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    <img src={avatar} width={24} height={24} />
+  ) : (
+    <Unicon size={24} address={asset.ownerAddress ?? ''} />
+  )
+
+  return (
+    <Column gap="sm">
+      <InfoChipsContainer justify="center">
+        <InfoBubble key="Owner" title="Owner" info={addressToDisplay} icon={avatarToDisplay} />
+        <InfoBubble key="Trait Floor" title="Trait Floor" info="5.3 ETH" icon={<DollarSign size={20} />} />
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:109]: Design used a custom dollar sign with an underline in the Figma 
Screen Shot 2023-05-12 at 14 04 18

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  const addressToDisplay = ENSName ?? shortenedAddress
+  const avatarToDisplay = avatar ? (
+    <img src={avatar} width={24} height={24} />
+  ) : (
+    <Unicon size={24} address={asset.ownerAddress ?? ''} />
+  )
+
+  return (
+    <Column gap="sm">
+      <InfoChipsContainer justify="center">
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:107]: nit: you use justify center both places for InfoChipsContainer, I'd add it to the styles in the styled component

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+`
+
+export const InfoChips = ({ asset }: { asset: GenieAsset }) => {
+  const screenSize = useScreenSize()
+  const isMobile = !screenSize['sm']
+  const [showExtraInfoChips, setShowExtraInfoChips] = useState(false)
+  const shouldShowExtraInfoChips = isMobile && showExtraInfoChips
+
+  const topTrait = asset.traits && asset.traits.length ? asset.traits[0] : undefined
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:91]: I think you can shorten this to `const topTrait = asset?.traits?.[0]`

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  @media screen and (min-width: ${BREAKPOINTS.sm}px) {
+    gap: 12px;
+  }
+`
+
+export const InfoChips = ({ asset }: { asset: GenieAsset }) => {
+  const screenSize = useScreenSize()
+  const isMobile = !screenSize['sm']
+  const [showExtraInfoChips, setShowExtraInfoChips] = useState(false)
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:88]: nit: would be simpler as `const [showExtraInfoChips, toggleShowExtraInfoChips] = useReducer((s) => !s, false)`

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  height: 64px;
+
+  @media screen and (min-width: ${BREAKPOINTS.sm}px) {
+    gap: 12px;
+  }
+`
+
+export const InfoChips = ({ asset }: { asset: GenieAsset }) => {
+  const screenSize = useScreenSize()
+  const isMobile = !screenSize['sm']
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:87]: you can just use `isMobile()` to remove a line

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  height: 24px;
+  flex-shrink: 0;
+  color: ${({ theme }) => theme.accentAction};
+  border-radius: 100%;
+  overflow: hidden;
+  justify-content: center;
+  align-items: center;
+`
+
+const InfoBubble = ({ title, info, icon }: { title: string; info: string; icon: ReactNode }) => {
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:46]: Make title a ReactNode and pass the string wrapped in `Trans`

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+`
+
+const StyledLabelMedium = styled.div`
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 20px;
+  color: ${({ theme }) => theme.textPrimary};
+  text-overflow: ellipsis;
+  overflow: hidden;
+  white-space: nowrap;
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:32]: There's a shared style `EllipsisStyle` you can use instead of these three

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+`
+
+const StyledIcon = styled(Row)`
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:35]: Does this need to be a row with justify and align if it only contains 1 item?

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  return (
+    <Column gap="sm">
+      <InfoChipsContainer justify="center">
+        <InfoBubble key="Owner" title="Owner" info={addressToDisplay} icon={avatarToDisplay} />
+        <InfoBubble key="Trait Floor" title="Trait Floor" info="5.3 ETH" icon={<DollarSign size={20} />} />
+        {!isMobile && topTrait && <InfoBubble key="Top Trait" title="Top Trait" info={topTrait.trait_value} icon="" />}
+        {topTrait && (
+          <InfoChipDropdownContainer>
+            <InfoChipDropdown onClick={() => setShowExtraInfoChips(!showExtraInfoChips)}>
+              {showExtraInfoChips ? <ChevronUp size={20} display="block" /> : <ChevronDown size={20} display="block" />}
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:114]: I would replace the use of two components with a transform to rotate 180 and include an animation

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+            </InfoChipDropdown>
+          </InfoChipDropdownContainer>
+        )}
+      </InfoChipsContainer>
+      {shouldShowExtraInfoChips && topTrait && (
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:119]: you can accomplish this without declaring the same item twice, check this PR which leverages `flex-wrap`

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+  const avatarToDisplay = avatar ? (
+    <img src={avatar} width={24} height={24} />
+  ) : (
+    <Unicon size={24} address={asset.ownerAddress ?? ''} />
+  )
+
+  return (
+    <Column gap="sm">
+      <InfoChipsContainer justify="center">
+        <InfoBubble key="Owner" title="Owner" info={addressToDisplay} icon={avatarToDisplay} />
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:108]: nit: should this bubble be clickable and take me to the user's etherscan link?

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        <InfoBubble key="Trait Floor" title="Trait Floor" info="5.3 ETH" icon={<DollarSign size={20} />} />
+        {!isMobile && topTrait && <InfoBubble key="Top Trait" title="Top Trait" info={topTrait.trait_value} icon="" />}
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:110]: nit: should this bubble be clickable and take me to the collection page with a filter for this trait, similar to what we do with Trait rows in the Trait Table?

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/nft/components/details/detailsV2/InfoChips.tsx (hunk 1 above)
Comment (JackShort) [reply src/nft/components/details/detailsV2/InfoChips.tsx:118]: they said its fine to just use the one from react-feather

File: src/nft/components/details/detailsV2/InfoChips.tsx (hunk 8 above)
Comment (JackShort) [reply src/nft/components/details/detailsV2/InfoChips.tsx:35]: it aligns the svg in the center because the size of the svg is smaller than `StyledIcon`

File: src/nft/components/details/detailsV2/InfoChips.tsx [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+  info: string
+  icon: ReactNode
+  href: string
+  isExternal?: boolean
+}) => {
+  return (
+    <ConditionalLinkWrapper isExternal={isExternal} href={href}>
+      <Column gap="sm">
+        <ThemedText.Caption color="textSecondary">{title}</ThemedText.Caption>
+        <StyledBubble gap="sm">
```
Comment (cbachmeier) [inline src/nft/components/details/detailsV2/InfoChips.tsx:89]: nit: I think only the bubble should be clickable and have the hover state but fine either way
Screen Shot 2023-05-15 at 09 00 47

**Review by cbachmeier — APPROVED** [review]
LGTM, minor ux preference

File: src/nft/components/details/detailsV2/InfoChips.tsx (hunk 13 above)
Comment (JackShort) [reply src/nft/components/details/detailsV2/InfoChips.tsx:89]: good callout

**Review by zzmp — APPROVED** [review]
